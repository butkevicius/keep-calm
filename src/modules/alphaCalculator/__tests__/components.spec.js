import React from 'react';
import Components from '../components';

const { Overlay, Input, Submit, Form } = Components;

describe('AlphaCalculator components', () => {
    describe('Overlay', () => {
        it('should render', () => {
            const wrapper = shallow(<Overlay value={1} close={() => {}}/>);

            expect(wrapper).toMatchSnapshot();
        });

        it('should display given value', () => {
            const wrapper = shallow(<Overlay value={25} close={() => {}}/>);
            expect(wrapper.text()).toContain(25);
        });

        it('should invoke callback on button click', () => {
            const cb = jest.fn();
            const wrapper = shallow(<Overlay value={25} close={cb}/>);
            wrapper.find('button').simulate('click');
            expect(cb).toHaveBeenCalled();
        });
    });

    describe('Input', () => {
        it('should render', () => {
            const wrapper = shallow(<Input value={'1'} onChange={() => {}}/>);

            expect(wrapper).toMatchSnapshot();
        });

        it('should display given value', () => {
            const wrapper = shallow(<Input value={'25'} onChange={() => {}}/>);
            expect(wrapper.find('input').props().value).toContain('25');
        });

        it('should invoke callback on change', () => {
            const cb = jest.fn();
            const wrapper = shallow(<Input value={'25'} onChange={cb}/>);
            wrapper.find('input').simulate('change', {target: {value: '12345'}});
            expect(cb).toHaveBeenCalledWith('12345');
        });
    });

    describe('Submit', () => {
        it('should render with isEnabled true', () => {
            const wrapper = shallow(<Submit isEnabled submit={() => {}}/>);

            expect(wrapper).toMatchSnapshot();
        });

        it('should render with isEnabled false', () => {
            const wrapper = shallow(<Submit isEnabled={false} submit={() => {}}/>);

            expect(wrapper).toMatchSnapshot();
        });

        it('should submit if is enabled', () => {
            const cb = jest.fn();
            const wrapper = shallow(<Submit isEnabled submit={cb}/>);

            wrapper.find('button').simulate('click');
            expect(cb).toHaveBeenCalled();
        });

        it('should not submit if is disabled', () => {
            const cb = jest.fn();
            const wrapper = shallow(<Submit isEnabled={false} submit={cb}/>);

            wrapper.find('button').simulate('click');
            expect(cb).toHaveBeenCalledTimes(0);
        });
    });

    describe('Form', () => {
        it('should render', () => {
            const wrapper = mount(<Form />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});