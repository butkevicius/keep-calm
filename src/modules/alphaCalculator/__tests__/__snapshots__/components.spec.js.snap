exports[`AlphaCalculator components Form should render 1`] = `
<Form>
  <div
    className="AlphaCalculatorForm">
    <Input>
      <input
        className="AlphaCalculatorInput"
        onChange={[Function]}
        placeholder="Enter a number"
        type="text" />
    </Input>
    <Submit>
      <div
        className="AlphaCalculatorSubmit is-disabled">
        <button
          disabled={true}
          onClick={[Function]}>
          Get Calculations
        </button>
      </div>
    </Submit>
  </div>
</Form>
`;

exports[`AlphaCalculator components Input should render 1`] = `
<input
  className="AlphaCalculatorInput"
  onChange={[Function]}
  placeholder="Enter a number"
  type="text"
  value="1" />
`;

exports[`AlphaCalculator components Overlay should render 1`] = `
<div
  className="AlphaCalculatorOverlay">
  <div
    className="AlphaCalculatorOverlay-content">
    <div
      className="AlphaCalculatorOverlay-text">
      Result: 
      1
    </div>
    <div
      className="AlphaCalculatorOverlay-controls">
      <button
        className="AlphaCalculatorOverlay-button"
        onClick={[Function]}>
        Enter new value
      </button>
    </div>
  </div>
</div>
`;

exports[`AlphaCalculator components Submit should render with isEnabled false 1`] = `
<div
  className="AlphaCalculatorSubmit is-disabled">
  <button
    disabled={true}
    onClick={[Function]}>
    Get Calculations
  </button>
</div>
`;

exports[`AlphaCalculator components Submit should render with isEnabled true 1`] = `
<div
  className="AlphaCalculatorSubmit ">
  <button
    disabled={false}
    onClick={[Function]}>
    Get Calculations
  </button>
</div>
`;
